core/src/main/java/org/apache/iceberg/MicroBatches.java

```diff
@@ -92,7 +92,7 @@ private static List<Pair<ManifestFile, Integer>> indexManifests(
 
     for (ManifestFile manifest : manifestFiles) {
       manifestIndexes.add(Pair.of(manifest, currentFileIndex));
-      currentFileIndex += manifest.addedFilesCount() + manifest.existingFilesCount();
+      currentFileIndex += manifest.addedFilesCount();
```

Comment:
I would but I don't know how?

Reply:
ya we need to think of a case where this would actually result in incorrect results i am a bit surprised that there is no existing case for testing this code path. when i was adding rate limit code i just refactored this part to a common function.

Reply:
ya none of the snapshots have an `existing file count` value.

when I look at production tables it's pretty rare actually to see that. I don't know what makes the `existing file count` be set.

Reply:
imho then we should not remove this line until we have coverage for this then, can you please revert this change then ?

Reply:
I think it is relatively easy to reason about this part because it is used only in one path.

The existing entries will be entries of data files that are already part of a manifest. For example, when you rewrite the metadata; for example because you use a lot of fast-appends. With a fast append, a new manifest will be written and added to the manifest-list. At some point, you want to combine these manifests into a bigger one to reduce the number of calls to the object store and make the planning part faster.

We want to skip when only the metadata has been rewritten, so I think this change is correct.

Reply:
Exactly, see here I show how the existingFilesCount is only set when doing rewrites.

https://github.com/apache/iceberg/pull/8980#issuecomment-1853964957

However the micro-batch streaming reader skips over rewrites. So it will never happen that the existingFilesCount is set in the above code.

Reply:
It can happen, for example, it is easy to reproduce in `TestTransaction::testTransactionRecommit` where a new datafile is committed, and merged into an existing one:

<img width="542" alt="image" src="https://github.com/apache/iceberg/assets/1134248/3cab9928-41c0-4fdc-b7b8-420f66ad906a">

Now, I'm questioning `skipManifests`, if it is valid to rely on the counts of manifests. But I have to dig deeper into this code since I'm not too familiar with it.

Reply:
@singhpk234  I'll put the ` + manifest.existingFilesCount();` back.

Now that I know how to create manifests with existing file counts I will add this to the test suite

`commit.manifest.min-count-to-merge=3`

This way the test will include some manifest entries with existing file counts. I ran the tests again with the ` + manifest.existingFilesCount();` and without. All the test still pass. But I think that makes sense because the `skipManifests` is an optimization.

Have a look and let me know what you think.